modules/monitoring/main.tf

```diff
@@ -0,0 +1,217 @@
+# Cloudwatch log group which hosts the logs
+
+resource "aws_cloudwatch_log_group" "graphdb_log_group" {
+  name              = "${var.resource_name_prefix}-graphdb"
+  retention_in_days = var.log_group_retention_in_days
+}
+
+# SSM Parameter which hosts the config for the cloudwatch agent
+
+resource "aws_ssm_parameter" "cloudwatch_agent_config" {
+  name        = "/CWAgent/Config"
+  description = "Cloudwatch Agent Configuration"
+  type        = "String"
+  tier        = "Advanced"
+
+  value = templatefile("${path.module}/cloudwatch_agent_config.json.tpl", { name : var.resource_name_prefix })
```

Comment:
This could be included in the userdata scripts as a separate step and use the output here

Reply:
There were Issues with the interpolation in the section where we use append dimensions, that's why I chose this approach. We spoke about it.